torch/lib/c10d/ProcessGroupGloo.hpp

```diff
@@ -69,12 +69,16 @@ class ProcessGroupGloo : public ProcessGroup {
   // operations using the new AsyncWork base class. Over time we will port
   // all operations and perform needed cleanup.
   //
+  // FIXME: This probably should be called WorkGloo since the work is executed in sync mode
+  // by a background thread.
   class AsyncWork : public ProcessGroup::Work {
    public:
     AsyncWork(
+        std::vector<std::vector<at::Tensor>> outputTensors,
         const char* profilingTitle = nullptr,
         const c10::optional<std::vector<at::Tensor>>& inputTensors = c10::nullopt)
-        : ProcessGroup::Work(-1, OpType::UNKNOWN, profilingTitle, inputTensors) {
+        : ProcessGroup::Work(-1, OpType::UNKNOWN, profilingTitle, inputTensors),
+          outputTensors_(outputTensors) {
```

Comment:
Shouldn't be `outputTensors_(std::move(outputTensors))`?

Otherwise it will create a separate vector copy. Although the underlying data of each tensor is shared, a copy of this vector can still be non-trivial, unless the length is always short.

Reply:
the intent is to create a separate vector copy, so even if user's output goes out of scope, e.g. user solely relies on result() function as in DDP comm hook code, the underlying tensors won't be going away and this won't result in a SEG.